chainer/functions/connection/n_step_lstm.py

```diff
@@ -0,0 +1,476 @@
+            hx_desc.value, hx.data.ptr, self.c_y_descs.data, self.ys.data.ptr,
+            workspace.data.ptr, work_size, dw_desc.value, dw.data.ptr,
+            self.reserve_space.data.ptr, self.reserve_space.size)
+
+        dx = dx_list[0]
+        dx = dx.reshape(dx.shape + (1,))
+        dx_desc = cudnn.create_tensor_nd_descriptor(dx)
+        dws = [cuda.cupy.empty_like(w) for w in ws]
+        dbs = [cuda.cupy.empty_like(b) for b in bs]
+        for layer in six.moves.range(self.n_layers):
+            for lin_layer_id in six.moves.range(8):
+                mat = cudnn.get_rnn_lin_layer_matrix_params(
+                    handle, rnn_desc, layer, dx_desc, dw_desc, dw,
+                    lin_layer_id)
+                v = dws[layer * 8 + lin_layer_id]
+                v = v.reshape(v.size)
+                v[:] = mat.ravel()
+                bias = cudnn.get_rnn_lin_layer_bias_params(
+                    handle, rnn_desc, layer, dx_desc, dw_desc, dw,
+                    lin_layer_id)
+                v = dbs[layer * 8 + lin_layer_id]
+                v = v.reshape(v.size)
+                v[:] = bias.ravel()
+
+        return tuple([dhx, dcx] + dws + dbs + dx_list)
+
+
+def _stack_weight(ws):
+    # TODO(unno): Input of the current LSTM implementaiton is shuffled
+    w = stack.stack(ws, axis=1)
+    shape = w.data.shape
+    return reshape.reshape(w, (shape[0] * shape[1],) + shape[2:])
+
+
+def n_step_lstm(
+        n_layers, dropout_ratio, hx, cx, ws, bs, xs, seed=1337, train=True,
+        use_cudnn=True):
+    """Stacked Long Short-Term Memory function for sequence inputs.
+
+    This function calculates stacked LSTM with sequences. This function gets
+    an initial hidden state :math:`h_0`, an initial cell state :math:`c_0`,
+    an input sequence :math:`x`, weight matrices :math:`W`, and bias vectors
+    :math:`b`.
+    This function calculates hidden states :math:`h_t` and :math:`c_t` for each
+    time :math:`t` from input :math:`x_t`.
+
+    .. math::
+
+       i_t = \sigma(W_0 x_t + W_4 h_{t-1} + b_0 + b_4)
+       f_t = \sigma(W_1 x_t + W_5 h_{t-1} + b_1 + b_5)
+       o_t = \sigma(W_2 x_t + W_6 h_{t-1} + b_2 + b_6)
+       a_t = \tanh(W_3 x_t + W_7 h_{t-1} + b_3 + b_7)
+       c_t = f_t \dot c_{t-1} + i_t \dot a_t
+       h_t = o_t \dot \tanh(c_t)
+
+    As the function accepts a sequence, it calculates :math:`h_t` for all
+    :math:`t` with one call. Eight weight matrices and eight bias vectors are
+    required for each layers. So, when :math:`S` layers exists, you need to
+    prepare :math:`8S` weigth matrices and :math:`8S` bias vectors.
+
+    If the number of layers ``n_layers`` is greather than :math:`1`, input
+    of ``k``-th layer is hidden state ``h_t`` of ``k-1``-th layer.
+    Note that all input variables except first layer may have different shape
+    from the first layer.
+
+    Args:
+        n_layers(int): Number of layers.
+        dropout_ratio(float): Dropout ratio.
```

Comment:
Can we specify what variables dropout is applied to? For example, it seems like the recurrent states `c` and `h` need a particular kind of dropout (http://arxiv.org/abs/1512.05287) that's slightly different from dropout as applied to the inputs `xs`